### PR TITLE
MIGENG-336: Alter helper functions to provide new isConvertible() met…

### PR DESCRIPTION
**Pull request by m-brophy**
…hod and update the logic of the others

https://issues.redhat.com/browse/MIGENG-336

### CODE REVIEW

File: src/main/java/org/jboss/xavier/analytics/functions/HelperFunctions.java [hunk 1]
```diff
@@ -16,24 +16,27 @@ public static boolean isSupportedOS(String osToCheck)
...
 
-    public static boolean isUndetectedOS(String osToCheck)
+    public static boolean isUnsupportedOS(String osToCheck)
     {
         return Arrays.stream(OSSupport.values()).noneMatch(value -> osToCheck.toLowerCase().contains(value.getName().toLowerCase()));
     }
 
+    public static boolean isUndetectedOS(String osToCheck)
+    {
+        return osToCheck == null || osToCheck.equals("");
```
Comment (carlosthe19916) [inline src/main/java/org/jboss/xavier/analytics/functions/HelperFunctions.java]: Just a suggestion:
```suggestion
        return osToCheck == null || osToCheck.trim().isEmpty();
```

**Review by carlosthe19916 — COMMENTED** [review]
@m-brophy The code looks good to me.

**Review by carlosthe19916 — APPROVED** [review]

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadInventoryReportTest.java [hunk 2]
```diff
@@ -441,7 +441,7 @@ public void testNoFlagsUnSupportedOS() throws ParseException {
         Assert.assertTrue(targets.contains("OSP"));
         Assert.assertTrue(targets.contains("Convert2RHEL"));
         // Complexity
-        Assert.assertEquals(WorkloadInventoryReportModel.COMPLEXITY_MEDIUM,workloadInventoryReportModel.getComplexity());
+        //Assert.assertEquals(WorkloadInventoryReportModel.COMPLEXITY_MEDIUM,workloadInventoryReportModel.getComplexity());
```
Comment (mrizzi) [inline src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadInventoryReportTest.java:444]: Is this test check going to be uncommented in one of the further PRs?
Maybe another PR will introduce again this check with the new and right complexity check?

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadInventoryReportTest.java [hunk 3]
```diff
@@ -545,7 +545,7 @@ public void testFlagsCentOS() throws ParseException {
         Assert.assertTrue(targets.contains("RHV"));
         Assert.assertTrue(targets.contains("Convert2RHEL"));
         // Complexity
-        Assert.assertEquals(WorkloadInventoryReportModel.COMPLEXITY_HARD,workloadInventoryReportModel.getComplexity());
+        //Assert.assertEquals(WorkloadInventoryReportModel.COMPLEXITY_HARD,workloadInventoryReportModel.getComplexity());
```
Comment (mrizzi) [inline src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadInventoryReportTest.java:548]: Same as the above comment

**Review by mrizzi — COMMENTED** [review]
Commented to know more about commented checks in some tests.

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadInventoryReportTest.java (hunk 2 above)
Comment (m-brophy) [reply src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadInventoryReportTest.java:444]: This check is reintorodeuced with a different value in MIGENG-337

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadInventoryReportTest.java (hunk 3 above)
Comment (m-brophy) [reply src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadInventoryReportTest.java:548]: This check is reintroduced with a different value in MIGENG-337

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadInventoryReportTest.java (hunk 2 above)
Comment (mrizzi) [reply src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadInventoryReportTest.java:444]: Great, thanks @m-brophy

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadInventoryReportTest.java (hunk 3 above)
Comment (mrizzi) [reply src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/WorkloadInventoryReportTest.java:548]: Great, thanks @m-brophy

**Review by mrizzi — APPROVED** [review]
This "preparatory" PR looks good to me, thanks @m-brophy :+1:

**Pull request merged**